react/lib/util/api-client.ts

```diff
@@ -10,7 +10,31 @@ import {
   Currency,
 } from './types';
 import { isFiat } from './currency';
+import { resolveNumber } from './number';
 
+export const shouldTriggerOnSuccess = (
+  transaction: Transaction,
+  thisPaymentId?: string,
+  cryptoAmount?: string,
+  thisOpReturn?: string,
+) => {
+  const { amount, paymentId, message } = transaction;
+
+  const formattedTxAmount = resolveNumber(amount);
+  const formattedAmountSet = cryptoAmount ? resolveNumber(cryptoAmount) : false;
```

Comment:
Maybe call this `formattedExpectedAmount`, or something like that. This function in general is being quite hard to read — the bottom line here  is that there are values that come in the tx and there are values that had been defined in the paybutton, and we want to trigger onSuccess if they match.

A variable of type `BigNumber | boolean` named with a suffix ("Set") that only refers to one of the possible types (the boolean one), conjugated in the past tense, which is a homograph of the infinitive (also "set"),  without expliciting that this refers to the expected value — certainly doesn't help, especially when this suffix is [the word with most meanings in the English language](https://www.guinnessworldrecords.com/world-records/english-word-with-the-most-meanings/).

Reply:
thanks for the feedback, I just gave it a little cleanup to improve reading and also make it less verbose